commons/all/src/main/java/org/infinispan/commons/dataconversion/DefaultTranscoder.java

```diff
@@ -109,6 +110,7 @@ public Object convertToOctetStream(Object content, MediaType contentType, MediaT
          if (content instanceof String) {
             return content.toString().getBytes(UTF_8);
          }
+         if (content instanceof WrappedByteArray) return ((WrappedByteArray) content).getBytes();
```

Comment:
I updated the `EventListenerKeysFilter` to not use the `DataConversion` internally and let the notifier do the parsing. But then it started failing with a simple cache because the keys were WBA.

Reply:
This is normally handled by the wrapper portion of the DataConversion. Is that being skipped somewhere?

Reply:
I think I got it. The failure is on the test running with a simple cache. The encoder does not register with all the information:

https://github.com/infinispan/infinispan/blob/8cfa647b84f497b3837ca8682b3b32eaa621c2a8/core/src/main/java/org/infinispan/cache/impl/EncoderCache.java#L1060-L1069

Which uses the identity conversion:

https://github.com/infinispan/infinispan/blob/8cfa647b84f497b3837ca8682b3b32eaa621c2a8/core/src/main/java/org/infinispan/notifications/cachelistener/CacheNotifierImpl.java#L1184-L1187

We can get away by checking the instance on the filter. But maybe something we want to change to pass the encoded info to the notifier on another jira.